lib/razorpay/request.rb

```diff
@@ -115,5 +115,18 @@ def raise_error(error, status)
       # We got an unknown error, cast it to Error for now
       raise Razorpay::Error.new, 'Unknown Error'
     end
+
+    def entity_check(entity)
+      child_classes = ObjectSpace.each_object(Class).select {|klass| klass < Razorpay::Entity }
```

Comment:
We can also expose this list of entities as constant and encourage merchants to use the same, something like Razorpay::Generic.new(Razorpay::ORDER_ENTITY) instead of Razorpay::Generic.new("orders")